File: listtryjava/src/main/java/com/codekul/listtryjava/MainActivity.java
package com.codekul.listtryjava;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

//Don't fucking declear your Veriable private over here use final next time onCreate method

    private int elementPos =0 ;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final List<String> arrayList = new ArrayList< >();
        final ListView list = (ListView)findViewById(R.id.lst);
        final TextView tx =  findViewById(R.id.et);

        arrayList.add("One");
        arrayList.add("Two");
        arrayList.add("Three");
        arrayList.add("Four");
        arrayList.add("Five");
        arrayList.add("Six");
        arrayList.add("Seven");


        final ArrayAdapter<String> adapter = new ArrayAdapter<>(
                this, android.R.layout.simple_list_item_1, arrayList);
        list.setAdapter(adapter);

        (findViewById(R.id.btnOk)).setOnClickListener(

                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        ((ArrayAdapter<String>) list.getAdapter()).add((tx.getText().toString()));
                        ((ArrayAdapter<String>) list.getAdapter()).notifyDataSetChanged();
                        tx.setText("");
                    }
                });

        (findViewById(R.id.btnUpdate)).setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        ((ArrayAdapter<String>) list.getAdapter()).remove(arrayList.get(elementPos));
                        ((ArrayAdapter<String>) list.getAdapter()).insert(tx.getText().toString(),elementPos);
                        ((ArrayAdapter<String>) list.getAdapter()).notifyDataSetChanged();
                        tx.setText("");
                    }
                }
        );


        list.setOnItemClickListener(
                new AdapterView.OnItemClickListener() {
                    @Override
                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                        tx.setText(arrayList.get(position));
                        elementPos = position;
                    }
                }
        );

        list.setOnItemLongClickListener(
                new AdapterView.OnItemLongClickListener(){
                    @Override
                    public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
                    adapter.remove(adapter.getItem(position));
                        return true;
                    }
                }
        );

    }



}

File: listtrykotlin/src/main/java/com/codekul/listtrykotlin/MainActivity.kt
package com.codekul.listtrykotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.Toast
import com.codekul.listtrykotlin.R.id.lst
import kotlinx.android.synthetic.main.activity_main.*
var rawDT : ArrayList<String> = ArrayList()
var  elementPos : Int = 0

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

         rawDT.add("ONE")
        rawDT.add("TWO")
        rawDT.add("THREE")
        rawDT.add("FOUR")
        rawDT.add("FIVE")
        rawDT.add("SIX")
        var adapter = ArrayAdapter<String>(this@MainActivity,android.R.layout.simple_list_item_1,rawDT)
        lst.adapter = adapter

        btnOk.setOnClickListener(this::added) // you can put method outside using such syntax " :: " but don't forget to pass View as Constructor


        //list onListener
        lst.setOnItemClickListener { parent, view, position, id -> et.setText(rawDT.get(position))
        Toast.makeText(this,"Touched is : "+rawDT.get(position) ,Toast.LENGTH_LONG).show()
            elementPos = position
        } // "_" to mute veriable if we are not using veriabl;e just mute it kotlin will handle other things

        lst.setOnItemLongClickListener { parent, view, position, id ->
            Toast.makeText(this,"removed element : "+rawDT.get(position) ,Toast.LENGTH_LONG).show()
            adapter.remove(adapter.getItem(position))
            true }

  // updated data
        btnUpdate.setOnClickListener(this :: updated)
    }

  internal  fun  added(vw : View){ //function can be Internal it's almost same as Private
      (lst.adapter as ArrayAdapter<String>).add(et.text.toString())
      (lst.adapter as ArrayAdapter<String>).notifyDataSetChanged() //if ui is not getting updated use this
      et.setText("")

    }
    internal  fun updated(vw : View){
        (lst.adapter as ArrayAdapter<String>).remove(rawDT.get(elementPos))
        (lst.adapter as ArrayAdapter<String>).insert((et.text.toString()), elementPos)
        (lst.adapter as ArrayAdapter<String>).notifyDataSetChanged() //if ui is not getting updated use this
        et.setText("")
    }



}
